### PR TITLE
Add (some) evil vterm bindings

### PR DESCRIPTION
**Pull request by mjlbach**
See https://github.com/akermu/emacs-libvterm/pull/455 and https://github.com/akermu/emacs-libvterm/issues/313#issuecomment-738842507

We should probably add few more... Suggestions?

Edit: This isn't actually editing buffer contents and is broken atm

### CODE REVIEW

**Comment by mjlbach** [comment]
*draft, some things aren't working

File: modules/term/vterm/config.el [hunk 1]
```diff
@@ -26,4 +26,34 @@
...
+
+      ;; Add evil specific bindings that work with vterm mode
+      (defun vterm-evil-insert ()
+        (interactive)
+        (vterm-goto-char (point))
+        (call-interactively #'evil-insert))
+      (defun vterm-evil-append ()
+        (interactive)
+        (vterm-goto-char (point))
+        (call-interactively #'evil-append))
```
Comment (jixiuf) [inline modules/term/vterm/config.el:39]: this should be （1+ （point））

**Comment by mjlbach** [comment]
Thanks @jixiuf! (I copied from you, so not sure how that happened). I'm not quite sure why, but the delete bindings are still not working (deletes in rendered buffer but not on the cli)

I added this to my zshrc
```
vterm_printf(){
    if [ -n "$TMUX" ]; then
        # Tell tmux to pass the escape sequences through
        # (Source: http://permalink.gmane.org/gmane.comp.terminal-emulators.tmux.user/1324)
        printf "\ePtmux;\e\e]%s\007\e\\" "$1"
    elif [ "${TERM%%-*}" = "screen" ]; then
        # GNU screen (screen, screen-256color, screen-256color-bce)
        printf "\eP\e]%s\007\e\\" "$1"
    else
        printf "\e]%s\e\\" "$1"
    fi
}

vterm_prompt_end() {
    vterm_printf "51;A$(whoami)@$(hostname):$(pwd)";
}
setopt PROMPT_SUBST
PROMPT=$PROMPT'%{$(vterm_prompt_end)%}'
```

**Comment by jixiuf** [comment]
That‘s intended behavior. Only the area after last prompt are editable, you can only delete/change there.

**Comment by mjlbach** [comment]
Yes, it wasn't working on the last prompt, but it seems to be fine now.

**Pull request closed without merging**